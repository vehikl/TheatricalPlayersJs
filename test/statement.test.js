const statement = require('../src/statement');
const fs=require('fs');

test('example statement', () => {
    const invoice = JSON.parse(fs.readFileSync('test/invoice.json', 'utf8'));
    const plays = JSON.parse(fs.readFileSync('test/plays.json', 'utf8'));
    expect(statement(invoice, plays)).toMatchSnapshot();
});

test('example html statement', () => {
    const invoice = JSON.parse(fs.readFileSync('test/invoice.json', 'utf8'));
    const plays = JSON.parse(fs.readFileSync('test/plays.json', 'utf8'));
    expect(statement(invoice, plays, 'html')).toMatchSnapshot();
})

test('statement with new play types', () => {
    const invoice = JSON.parse(fs.readFileSync('test/invoice_new_plays.json', 'utf8'));
    const plays = JSON.parse(fs.readFileSync('test/new_plays.json', 'utf8'));
    expect(() => {statement(invoice, plays)}).toThrow(/unknown type/);
});
