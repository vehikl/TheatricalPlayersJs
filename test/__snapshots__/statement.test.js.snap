// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`example statement 1`] = `
"Statement for BigCo
 Hamlet: $650.00 (55 seats)
 As You Like It: $580.00 (35 seats)
 Othello: $500.00 (40 seats)
Amount owed is $1,730.00
You earned 47 credits
"
`;

exports[`generates html statement 1`] = `
"<div>
 <h1>Statement for BigCo</h1>
 <b>Hamlet:</b> $650.00 (55 seats)<br>
 <b>As You Like It:</b> $580.00 (35 seats)<br>
 <b>Othello:</b> $500.00 (40 seats)<br>
Amount owed is $1,730.00<br>
You earned 47 credits
</div>"
`;
